### DISCUSSION TITLE
Gzhttp not supported decompress (Content-Encoding) request body

### DISCUSSION THREAD

**Discussion by mirecl** [klauspost/compress#1001] [Ideas/Features]
**Gzhttp** not supported uncompress (_Content-Encoding_) request body.
**Gzhttp** working only response body: https://github.com/klauspost/compress/blob/v1.17.9/gzhttp/compress.go#L467

**Comment by klauspost** [comment]
@mirecl Could you elaborate a bit more? Preferably with an example.

Note that you should not add `Accept-Encoding` if you want this package to decompress. If you add the accept upstream, it is assumed that upstream will decompress. [code](https://github.com/klauspost/compress/blob/v1.17.9/gzhttp/compress.go#L762).

**Comment by mirecl** [comment]
@klauspost, I send http-request with gzip body and set header ``Content-Encoding: gzip``, for example:
```sh
cat data.json | gzip | curl -v -i --data-binary @- -H "Content-Encoding: gzip" http://...
```
But https://github.com/klauspost/compress/gzhttp don`t decompress request body automatically(
I think that this functionality will be useful and I’m ready implement a PR!

_For more detail: https://datatracker.ietf.org/doc/html/rfc7694_

**Comment by klauspost** [comment]
@mirecl The package only does Response compression and decompression. Moving this to a feature request (pun not intended ;)

**Comment by mirecl** [comment]
@klauspost create for example implement this feature: https://github.com/klauspost/compress/pull/1002

**Comment by mirecl** [comment]
@klauspost, when to expect a new release with this feature?

**Reply by klauspost** [reply]
It does seem like there is a bunch of smaller things waiting for release. I will push one now.

**Reply by klauspost** [reply]
(but you can always use the latest commit hash in your go.mod ie `github.com/klauspost/compress v1.17.9` -> `github.com/klauspost/compress 2a46d6bf5d0fb5d9f44b815438ce43470706f73f` + `go mod tidy` will work just fine)
